opencti-platform/opencti-front/src/private/components/workspaces/Workspaces.tsx

```diff
@@ -46,23 +57,106 @@ const Workspaces: FunctionComponent<WorkspacesProps> = ({
 
-  const filters = {
-    mode: 'and',
-    filters: [{
-      key: ['type'],
-      values: [type],
-      mode: 'or',
-      operator: 'eq',
-    }],
-    filterGroups: [],
-  } as GqlFilterGroup;
-  const workspacePaginationOptions = { ...paginationOptions, filters };
+  const filters = useBuildEntityTypeBasedFilterContext(
+    'Workspace',
+    {
+      mode: 'and',
+      filters: [{
+        key: 'type',
+        values: [type],
+        mode: 'or',
+        operator: 'eq',
+      }],
+      filterGroups: viewStorage.filters ? [viewStorage.filters] : [],
+    },
+  );
+
+  const workspacePaginationOptions = {
+    ...paginationOptions,
+    filters,
+  } as unknown as WorkspacesLinesPaginationQuery$variables;
 
   const queryRef = useQueryLoading<WorkspacesLinesPaginationQuery>(
     workspacesLinesQuery,
     workspacePaginationOptions,
   );
 
+  const renderDataTable = () => {
+    const dataColumns: DataTableProps['dataColumns'] = {
+      name: {
+        id: 'name',
+        flexSize: 33,
+        render: ({ name }, h) => textInTooltip(name, h),
+      },
+      tags: {
+        id: 'tags',
+      },
+      owner: {
+        id: 'owner',
+      },
+      created_at: {
+        id: 'created_at',
+        flexSize: 16,
+      },
+      updated_at: {
+        id: 'updated_at',
+        flexSize: type === 'dashboard' ? 16 : 24,
+      },
+      ...(type === 'dashboard' ? {
+        isShared: {
+          id: 'isShared',
+        },
+      } : {}),
+    };
+
+    return queryRef && (
+      <DataTable
+        dataColumns={dataColumns}
+        isNavigable={true}
+        resolvePath={(data: WorkspacesLines_data$data) => {
+          return data.workspaces?.edges?.map((n) => n?.node);
+        }}
+        storageKey={LOCAL_STORAGE_KEY}
+        initialValues={initialStorageValues}
+        toolbarFilters={filters}
+        preloadedPaginationProps={{
+          linesQuery: workspacesLinesQuery,
+          linesFragment: workspacesLineFragment,
+          queryRef,
+          nodePath: ['workspaces', 'pageInfo', 'globalCount'],
+          setNumberOfElements: storageHelpers.handleSetNumberOfElements,
+        }}
+        lineFragment={workspaceLineFragment}
+        exportContext={{ entity_type: 'Workspace' }}
+        searchContextFinal={{ entityTypes: ['Workspace'] }}
+        additionalHeaderButtons={[
+          <ToggleButton key="cards" value="lines" aria-label="lines">
+            <Tooltip title={t_i18n('Lines view')}>
+              <ViewListOutlined color="primary" fontSize="small"/>
+            </Tooltip>
+          </ToggleButton>,
+        ]}
+        createButton={isFeatureEnable('FAB_REPLACEMENT') && (
+          <Security needs={[EXPLORE_EXUPDATE, INVESTIGATION_INUPDATE]}>
+            <WorkspaceCreation
+              paginationOptions={workspacePaginationOptions}
+              type={type}
+            />
+          </Security>
+        )}
+        taskScope={type === 'dashboard' ? 'DASHBOARD' : 'INVESTIGATION'}
```

Comment:
What is the difference between type 'dashboard' and  'DASHBOARD'

Reply:
dashboard = a property of an entity `Workspace`, could be `dashboard` or `investigation`
DASHBOARD = the scope of the background task that is declared in our graphql